### DISCUSSION TITLE
Streaming support for voice chat

### DISCUSSION THREAD

**Discussion by PavelNikolaichev** [open-webui/open-webui#4569] [General]
I think it might be a good idea to have support for streaming LLM response during the voice chat mode. I think it would improve overall feedback loop for end user.

One of the possible solutions might be smth like [voicechat2](https://github.com/lhl/voicechat2?tab=readme-ov-file), which utilizes sockets to constantly check for a stream response from LLM. It sends the response sentence by sentence or so.

If you have anything to ask or add, feel free to do so!

**Comment by PavelNikolaichev** [comment]
Gonna implement this one myself

**Reply by nathanhere** [reply]
interested in this too!

**Comment by TecReaGroup** [comment]
> Gonna implement this one myself

Hope soon! Thanks! It's really useful!

**Comment by danielcederberg** [comment]
Want this too!

**Comment by PavelNikolaichev** [comment]
Hi, sorry guys, I forgot completely to leave a comment on it, this stuff was not relevant, that's why I haven't implement it. You can check the debug's on the frontend and see that's voice input does already streams. If you want voice only mode or smth, you might want to try adding VAD before or after audio preprocessing. Shouldn't be too hard to configure.
